### ISSUE TITLE
After updating MA, Music Playlist now only play a couple of seconds of each song.

### ISSUE THREAD

**Issue by michguy2000** [music-assistant/hass-music-assistant#2927]
### What version of Music Assistant has the issue?

Server version 2.2.4

### What version of the Home Assistant Integration have you got installed?

2024.9.1

### Have you tried everything in the Troubleshooting FAQ and reviewed the Open and Closed Issues and Discussions to resolve this yourself?

- [X] Yes

### The problem

I updated MA to 2024.9.0 as well as MA server to 2.2.4 48hours ago.  Since then, previously created playlist from my Synololgy NAS or YouTube Music will only play a couple of seconds of each song.  However, if I select an Album instead of a playlist from either source, it will play through all the songs.

### How to reproduce

Either by script or going directly into MA, select a playlist from a local share or from YouTube music.

### Music Providers

Filesystem (remote share)-SynologyDS923+
YouTube Music

### Player Providers

Chromecast
Home Assistant MediaPlayers
Universal Group Player
UPnP/DLNA Player provider

### Full log output

see attatched
[MA Log Output.txt](https://github.com/user-attachments/files/17007275/MA.Log.Output.txt)

### Additional information

The troubleshooting I've done:
1. Reboot (not restart) Home Assistant
2. Test manually executing playlist from my local share and from YouTube Music.
3. Verified that the playlist from YouTube **DO** play correctly when played through the **yTube_music_player**.  But when played through MA, the problem does occur.
4. Upgraded MA to the just released 2024.9.1

### What version of Home Assistant Core are your running

2024.9.1

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Raspberry Pi

**Comment by OzGav** [comment]
Looks to be a duplicate of https://github.com/music-assistant/hass-music-assistant/issues/2917

**Comment by michguy2000** [comment]
Hello.  Thanks for the quick reply.

 

Judging from #2917 description, I wasn’t sure that was the same issue when I saw it in the forum before I created a new issue.  His issue was with playing M3u8 streams on the radio.   Mine is related to m3u playlist being streamed local or from YT.

 

Is the information in the Log I provided giving you this conclusion?

Just curious.

 

Also, (and I’ll update the case), I just downgraded MA in HACS to 2024.8.1.  Still have the problem.

Thanks and regards.

**Comment by michguy2000** [comment]
UPDATE:  I just downgraded MA in HACS to 2024.8.1.  Still have the problem.

**Comment by OzGav** [comment]
Changing the integration version will have no effect as this is a server issue. Yes the error in the log is the same

**Comment by michguy2000** [comment]
Okay. Thanks for the update.  I'll wait to hear from you guys when there's
a fix. Thanks again.

**Comment by Margravegit** [comment]
Experiencing the same issue: tracks skip on any queued list (playlists, albums,  artist). If I select a single track, it plays for a second, then restarts, and then plays normally afterwards.
Additionally, trying to stop the current play or shut down the player fails 9/10 times (play stops, then restarts or skips to the next song after a second)
Looking at the logs, seems like two queue streams start at almost the same time (variable delay but usually around 1 second), every time that anything is played

**Comment by OzGav** [comment]
@michguy2000 Do you have QUEUE FLOW MODE enabled?

**Comment by michguy2000** [comment]
I did when I first installed MA last year, but I have not seen the QUEUE Flow Mode option in the Player providers for several releases (updates).  Is there a new location for it?

Regards

**Comment by OzGav** [comment]
No 
![image](https://github.com/user-attachments/assets/3de79dd2-5f4f-48b4-97c2-15142f4d630b)

**Comment by johnboiles** [comment]
I had the same issue this morning. Deleting the player provider and re-adding fixed it for me. Details here: https://github.com/music-assistant/server/pull/1653#issuecomment-2353379499

**Comment by marcelveldt** [comment]
Please test again with the just released patch 2.2.5 (or 2.3.0b24 if you run beta)

**Comment by michguy2000** [comment]
Hey.  I updated to 2.2.5, rebooted, but the problem has not been fixed for me.  Playlists still only play a couple of seconds of each track.  Both from YouTube music and my local NAS.

**Comment by marcelveldt** [comment]
> Hey. I updated to 2.2.5, rebooted, but the problem has not been fixed for me. Playlists still only play a couple of seconds of each track. Both from YouTube music and my local NAS. 

This is not reproducable for me in any way so I need more details: The full log and I need to know exactly on what players this is happening and if it maybe works on some player type and not on another.

**Comment by michguy2000** [comment]
Thanks for the reply.

All of the speakers are in a group.  3 google mini, 2 google nest hub max, 2 google nest hubs and 1 google speaker.

The problem is occurring whether I use the group or a single speaker.

 

Also, I have a routine setup to play music when I arrive home.  When I got back from the store this morning, MA still had the issue of only playing a few seconds of a song.  But by the time I finished unloading the car, MA had reached the last song in the playlist and played it completely.   I just tested this again, this time using YouTube music playlist instead of from the NAS and same results.  Only played the last song in the playlist completely.  My head hurts from this. 😊

 

This is my first big issue I’ve posted.  When you say you need to see the “full log” what do you mean?  

The log I posted before was from MA > Settings > Core > Server Logging.  Is this what you are referring to or is it a different log?

Regards

**Comment by OzGav** [comment]
@michguy2000 do you have any players besides cast? If not then add the snapcast provider and streak to a browser tab.

is there anything unusual in your network setup or installation?

please confirm again that this is only affecting a playlist? Have you tried creating a new MA playlist and see if that is affected?

**Comment by michguy2000** [comment]
Hello.  

The only other speakers I have are Alexa Echo.

Yes, I just confirmed that it is only playlist being streamed from my NAS and YouTube music.  If I select an Album to play from either source, it plays all the songs.

I added snapcast player provider, but am not familiar with it.  I do not see how to use it in MA and stream to any player that is not shown in the “Players” selection.  No specific options for Snapcast to stream to web browser tab.

 

No. I have not setup anything out of the ordinary on my network.  It’s pretty straight forward.  HA and all devices are on a dedicated network.  No Vlans or anything like that.

 

The only other changes made today were updating the Alexa Media Player and Core update 2024.9.2.

And reboots after the Core update.

I’m currently running a synchronize on both YouTube Music and the Synology.  I had done this before which didn’t help so I’m not hopeful that it will make any difference.

Regards

**Comment by OzGav** [comment]
After adding the snapcast provider you just go to a web browser and type in YOUR_MA_IP_ADDRESS:1780 and then a page will appear and if you press the play butto you will then find a new speaker in MA and you can stream to it.

Also there some error in your supplied log that have been fixed in the most recent beta. If you can stop your current addon, install the beta, set that up and try it. If you have the probem still attach another log here

**Comment by michguy2000** [comment]
Good Morning.  I followed your Snapcast instructions and it worked.  Playlist streamed normal again.

Since the Playlists work when used in YouTube player and now Snapcast, I assume there is some setting for the Google/Nest players that got changed or corrupted during my updates in MA.

 

I’ve got a busy schedule today so later this evening, I’ll do a HA backup, uninstall and re-install Music Assistant to see if that corrects the issue.

 

If you have another idea or solution, let me know.

Thanks for all the assist.

Regards

**Comment by OzGav** [comment]
Rather than full nuclear you could intially try just removing the Cast provider, restart MA and then re-add as that will put the players back with default config.

**Comment by michguy2000** [comment]
🤣😂👌

That’s a great idea!!  

I am so used to dealing with stupid stuff related to Google products over the last couple of years that is usually my knee jerk reaction.   Anything Google/Nest related – Uninstall it, factory reset it, pray, re-install it, pray.

😀

**Comment by michguy2000** [comment]
Hey.  Things seem to be working correctly now!!

I took your suggestion

*	Removed the Cast Provider and also the Home Assistant MediaPlayers.
*	Rebooted.
*	Re-added Cast Provider and the Home Assistant MediaPlayers.
*	Added the Google/Nest speakers to Home Assistant MediaPlayers.
*	Enabled Queue Flow mode for the appropriate Players.

So far, so good!

 

Thanks for your help.  It’s much appreciated.

 

Sidenote – The only issue I had while testing was the music abruptly stopped on all speakers mid song.  The Nest hubs displayed connection lost and then announced I had to “link my account in the google home app”.  What are the odds of that happening when I’m trying to do a #!%^* test?

**Comment by OzGav** [comment]
Closing. Can reopen if the same problem recurs.

**Comment by johnboiles** [comment]
I just got the same issue again on 2.2.5 this morning. My playlist that plays via automation cycled through songs after only playing a handful of seconds. Eventually it stopped trying, I hit play in MA, and now things are back to normal. Here's my log

[music-assistant (3)-debug.log](https://github.com/user-attachments/files/17077816/music-assistant.3.-debug.log)

Strangely there's nothing in the log before `08:59:47.564`, even though the automation triggered at `8:52:46 AM`. I'm not sure why that would be; does MA do log rotation? The docker container shows it's been up for 3 days:
<img width="1085" alt="image" src="https://github.com/user-attachments/assets/1bb1c7a1-3983-4136-a07f-a71c20375aa2">

**Comment by johnboiles** [comment]
I got the same issue again today on 2.2.6. I wonder if it could be related to #2893 since it looks like it's something to do with the `ffmpeg` process. @OzGav can we reopen this?

My morning playlist would play only 2-21 seconds of each song. Here's the relevant part of the log:

```
2024-10-01 09:12:42.245 DEBUG (MainThread) [music_assistant.audio.media_stream] stream aborted (with code 255) for apple_music--4hmSTk5D://track/1651724072 - seconds streamed: 4.0
2024-10-01 09:12:42.254 VERBOSE (MainThread) [ffmpeg.4182] Process ffmpeg with PID 4182 stopped with returncode 255
```

Sometimes I get messages about multiple processes stopped.

```
2024-10-01 09:12:05.888 VERBOSE (MainThread) [ffmpeg.3942] Process ffmpeg with PID 3942 stopped with returncode 255
2024-10-01 09:12:05.889 VERBOSE (MainThread) [ffmpeg.3943] Process ffmpeg with PID 3943 stopped with returncode 255
2024-10-01 09:12:05.889 DEBUG (MainThread) [music_assistant.audio.media_stream] stream aborted (with code 255) for apple_music--4hmSTk5D://track/1703037761 - seconds streamed: 21.0
```

After restarting MA the issue went away.

<details><summary>Log snippet from the last few songs</summary>
<p>

```
2024-10-01 09:12:36.279 VERBOSE (MainThread) [music_assistant.streams] Got GET request to /single/media_player.main_stereo/53c016ed8f4c408387a19c33d639d3bb.mp3 from 192.168.2.31
headers: <CIMultiDictProxy('User-Agent': 'ESP32 HTTP Client/1.0', 'Host': '192.168.1.2:8097', 'Content-Length': '0')>

2024-10-01 09:12:36.331 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem Koets - Mooi (library://track/18257) to Main Stereo
2024-10-01 09:12:36.340 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4150
2024-10-01 09:12:36.340 VERBOSE (MainThread) [ffmpeg.4150] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -ac 2 -channel_layout stereo -ar 44100 -acodec pcm_f32le -f f32le -i - -af alimiter=limit=-2dB:level=false:asc=true,aresample=resampler=soxr:precision=30:osf=s16:dither_method=triangular_hp -f mp3 -b:a 320k -
2024-10-01 09:12:36.341 DEBUG (MainThread) [music_assistant.audio.media_stream] start media stream for: apple_music--4hmSTk5D://track/1690644641
2024-10-01 09:12:36.348 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4151
2024-10-01 09:12:36.349 VERBOSE (MainThread) [ffmpeg.4151] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -decryption_key efa69b6f30aefe382316568063be6270 -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 5xx,4xx -i <URL> -af volume=-6.03dB -acodec pcm_f32le -f f32le -ar 44100 -ac 2 -
2024-10-01 09:12:37.526 VERBOSE (MainThread) [ffmpeg.4151] Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'https://aod-ssl.itunes.apple.com/itunes-assets/Music126/v4/43/a3/d0/43a3d0b2-a30b-46f2-5e15-04c0bce531f5/mzaf_A1690644641.rmhq.aac.wa.mp4':
2024-10-01 09:12:37.526 VERBOSE (MainThread) [ffmpeg.4151] Metadata:
2024-10-01 09:12:37.526 VERBOSE (MainThread) [ffmpeg.4151] major_brand     : iso5
2024-10-01 09:12:37.527 VERBOSE (MainThread) [ffmpeg.4151] minor_version   : 1
2024-10-01 09:12:37.527 VERBOSE (MainThread) [ffmpeg.4151] compatible_brands: isomiso5hlsf
2024-10-01 09:12:37.527 VERBOSE (MainThread) [ffmpeg.4151] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:37.527 VERBOSE (MainThread) [ffmpeg.4151] Duration: 00:02:22.55, start: 0.000000, bitrate: 277 kb/s
2024-10-01 09:12:37.527 VERBOSE (MainThread) [ffmpeg.4151] Stream #0:0[0x1](und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 270 kb/s (default)
2024-10-01 09:12:37.528 DEBUG (MainThread) [ffmpeg.4151] Detected (input) content type: aac (aac)
2024-10-01 09:12:37.528 VERBOSE (MainThread) [ffmpeg.4151] Metadata:
2024-10-01 09:12:37.528 VERBOSE (MainThread) [ffmpeg.4151] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:37.528 VERBOSE (MainThread) [ffmpeg.4151] handler_name    : Core Media Audio
2024-10-01 09:12:37.528 VERBOSE (MainThread) [ffmpeg.4151] vendor_id       : [0][0][0][0]
2024-10-01 09:12:37.530 VERBOSE (MainThread) [ffmpeg.4151] Stream mapping:
2024-10-01 09:12:37.530 VERBOSE (MainThread) [ffmpeg.4151] Stream #0:0 -> #0:0 (aac (native) -> pcm_f32le (native))
2024-10-01 09:12:37.531 VERBOSE (MainThread) [ffmpeg.4151] Press [q] to stop, [?] for help
2024-10-01 09:12:37.539 VERBOSE (MainThread) [ffmpeg.4151] Output #0, f32le, to 'pipe:':
2024-10-01 09:12:37.540 VERBOSE (MainThread) [ffmpeg.4151] Metadata:
2024-10-01 09:12:37.540 VERBOSE (MainThread) [ffmpeg.4151] major_brand     : iso5
2024-10-01 09:12:37.540 VERBOSE (MainThread) [ffmpeg.4151] minor_version   : 1
2024-10-01 09:12:37.540 VERBOSE (MainThread) [ffmpeg.4151] compatible_brands: isomiso5hlsf
2024-10-01 09:12:37.541 VERBOSE (MainThread) [ffmpeg.4151] encoder         : Lavf60.16.100
2024-10-01 09:12:37.541 VERBOSE (MainThread) [ffmpeg.4151] Stream #0:0(und): Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s (default)
2024-10-01 09:12:37.541 VERBOSE (MainThread) [ffmpeg.4151] Metadata:
2024-10-01 09:12:37.541 VERBOSE (MainThread) [ffmpeg.4151] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:37.541 VERBOSE (MainThread) [ffmpeg.4151] handler_name    : Core Media Audio
2024-10-01 09:12:37.542 VERBOSE (MainThread) [ffmpeg.4151] vendor_id       : [0][0][0][0]
2024-10-01 09:12:37.542 VERBOSE (MainThread) [ffmpeg.4151] encoder         : Lavc60.31.102 pcm_f32le
2024-10-01 09:12:37.601 VERBOSE (MainThread) [ffmpeg.4150] Input #0, f32le, from 'fd:':
2024-10-01 09:12:37.601 VERBOSE (MainThread) [ffmpeg.4150] Duration: N/A, bitrate: 2822 kb/s
2024-10-01 09:12:37.601 VERBOSE (MainThread) [ffmpeg.4150] Stream #0:0: Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s
2024-10-01 09:12:37.604 VERBOSE (MainThread) [ffmpeg.4150] Stream mapping:
2024-10-01 09:12:37.604 VERBOSE (MainThread) [ffmpeg.4150] Stream #0:0 -> #0:0 (pcm_f32le (native) -> mp3 (libmp3lame))
2024-10-01 09:12:37.643 VERBOSE (MainThread) [ffmpeg.4150] Output #0, mp3, to 'pipe:':
2024-10-01 09:12:37.643 VERBOSE (MainThread) [ffmpeg.4150] Metadata:
2024-10-01 09:12:37.644 VERBOSE (MainThread) [ffmpeg.4150] TSSE            : Lavf60.16.100
2024-10-01 09:12:37.645 VERBOSE (MainThread) [ffmpeg.4150] Stream #0:0: Audio: mp3, 44100 Hz, stereo, s16p, 320 kb/s
2024-10-01 09:12:37.645 VERBOSE (MainThread) [ffmpeg.4150] Metadata:
2024-10-01 09:12:37.645 VERBOSE (MainThread) [ffmpeg.4150] encoder         : Lavc60.31.102 libmp3lame
2024-10-01 09:12:37.749 VERBOSE (MainThread) [ffmpeg.4151] Process ffmpeg with PID 4151 stopped with returncode 255
2024-10-01 09:12:37.750 DEBUG (MainThread) [music_assistant.audio.media_stream] stream aborted (with code 255) for apple_music--4hmSTk5D://track/1690644641 - seconds streamed: 4.0
2024-10-01 09:12:37.757 VERBOSE (MainThread) [ffmpeg.4150] Process ffmpeg with PID 4150 stopped with returncode 255
2024-10-01 09:12:37.799 VERBOSE (MainThread) [music_assistant.streams] Got GET request to /single/media_player.main_stereo/53c016ed8f4c408387a19c33d639d3bb.mp3 from 192.168.2.31
headers: <CIMultiDictProxy('User-Agent': 'ESP32 HTTP Client/1.0', 'Host': '192.168.1.2:8097', 'Content-Length': '0')>

2024-10-01 09:12:37.850 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem Koets - Mooi (library://track/18257) to Main Stereo
2024-10-01 09:12:37.857 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4166
2024-10-01 09:12:37.857 VERBOSE (MainThread) [ffmpeg.4166] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -ac 2 -channel_layout stereo -ar 44100 -acodec pcm_f32le -f f32le -i - -af alimiter=limit=-2dB:level=false:asc=true,aresample=resampler=soxr:precision=30:osf=s16:dither_method=triangular_hp -f mp3 -b:a 320k -
2024-10-01 09:12:37.858 DEBUG (MainThread) [music_assistant.audio.media_stream] start media stream for: apple_music--4hmSTk5D://track/1690644641
2024-10-01 09:12:37.865 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4167
2024-10-01 09:12:37.866 VERBOSE (MainThread) [ffmpeg.4167] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -decryption_key efa69b6f30aefe382316568063be6270 -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 5xx,4xx -i <URL> -af volume=-6.03dB -acodec pcm_f32le -f f32le -ar 44100 -ac 2 -
2024-10-01 09:12:39.147 VERBOSE (MainThread) [ffmpeg.4167] Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'https://aod-ssl.itunes.apple.com/itunes-assets/Music126/v4/43/a3/d0/43a3d0b2-a30b-46f2-5e15-04c0bce531f5/mzaf_A1690644641.rmhq.aac.wa.mp4':
2024-10-01 09:12:39.147 VERBOSE (MainThread) [ffmpeg.4167] Metadata:
2024-10-01 09:12:39.148 VERBOSE (MainThread) [ffmpeg.4167] major_brand     : iso5
2024-10-01 09:12:39.148 VERBOSE (MainThread) [ffmpeg.4167] minor_version   : 1
2024-10-01 09:12:39.148 VERBOSE (MainThread) [ffmpeg.4167] compatible_brands: isomiso5hlsf
2024-10-01 09:12:39.148 VERBOSE (MainThread) [ffmpeg.4167] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:39.148 VERBOSE (MainThread) [ffmpeg.4167] Duration: 00:02:22.55, start: 0.000000, bitrate: 277 kb/s
2024-10-01 09:12:39.149 VERBOSE (MainThread) [ffmpeg.4167] Stream #0:0[0x1](und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 270 kb/s (default)
2024-10-01 09:12:39.149 VERBOSE (MainThread) [ffmpeg.4167] Metadata:
2024-10-01 09:12:39.149 VERBOSE (MainThread) [ffmpeg.4167] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:39.149 VERBOSE (MainThread) [ffmpeg.4167] handler_name    : Core Media Audio
2024-10-01 09:12:39.150 VERBOSE (MainThread) [ffmpeg.4167] vendor_id       : [0][0][0][0]
2024-10-01 09:12:39.151 VERBOSE (MainThread) [ffmpeg.4167] Stream mapping:
2024-10-01 09:12:39.151 VERBOSE (MainThread) [ffmpeg.4167] Stream #0:0 -> #0:0 (aac (native) -> pcm_f32le (native))
2024-10-01 09:12:39.151 VERBOSE (MainThread) [ffmpeg.4167] Press [q] to stop, [?] for help
2024-10-01 09:12:39.158 VERBOSE (MainThread) [ffmpeg.4167] Output #0, f32le, to 'pipe:':
2024-10-01 09:12:39.158 VERBOSE (MainThread) [ffmpeg.4167] Metadata:
2024-10-01 09:12:39.158 VERBOSE (MainThread) [ffmpeg.4167] major_brand     : iso5
2024-10-01 09:12:39.158 VERBOSE (MainThread) [ffmpeg.4167] minor_version   : 1
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] compatible_brands: isomiso5hlsf
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] encoder         : Lavf60.16.100
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] Stream #0:0(und): Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s (default)
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] Metadata:
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] creation_time   : 2023-06-01T18:29:25.000000Z
2024-10-01 09:12:39.159 VERBOSE (MainThread) [ffmpeg.4167] handler_name    : Core Media Audio
2024-10-01 09:12:39.160 VERBOSE (MainThread) [ffmpeg.4167] vendor_id       : [0][0][0][0]
2024-10-01 09:12:39.160 VERBOSE (MainThread) [ffmpeg.4167] encoder         : Lavc60.31.102 pcm_f32le
2024-10-01 09:12:39.237 VERBOSE (MainThread) [ffmpeg.4166] Input #0, f32le, from 'fd:':
2024-10-01 09:12:39.237 VERBOSE (MainThread) [ffmpeg.4166] Duration: N/A, bitrate: 2822 kb/s
2024-10-01 09:12:39.237 VERBOSE (MainThread) [ffmpeg.4166] Stream #0:0: Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s
2024-10-01 09:12:39.239 VERBOSE (MainThread) [ffmpeg.4166] Stream mapping:
2024-10-01 09:12:39.240 VERBOSE (MainThread) [ffmpeg.4166] Stream #0:0 -> #0:0 (pcm_f32le (native) -> mp3 (libmp3lame))
2024-10-01 09:12:39.271 VERBOSE (MainThread) [ffmpeg.4166] Output #0, mp3, to 'pipe:':
2024-10-01 09:12:39.272 VERBOSE (MainThread) [ffmpeg.4166] Metadata:
2024-10-01 09:12:39.272 VERBOSE (MainThread) [ffmpeg.4166] TSSE            : Lavf60.16.100
2024-10-01 09:12:39.273 VERBOSE (MainThread) [ffmpeg.4166] Stream #0:0: Audio: mp3, 44100 Hz, stereo, s16p, 320 kb/s
2024-10-01 09:12:39.273 VERBOSE (MainThread) [ffmpeg.4166] Metadata:
2024-10-01 09:12:39.273 VERBOSE (MainThread) [ffmpeg.4166] encoder         : Lavc60.31.102 libmp3lame
2024-10-01 09:12:39.852 DEBUG (MainThread) [music_assistant.audio] Getting streamdetails for apple_music://track/1651724072
2024-10-01 09:12:40.531 DEBUG (MainThread) [music_assistant.audio] retrieved streamdetails for apple_music://track/1651724072 in 679 milliseconds
2024-10-01 09:12:40.732 VERBOSE (MainThread) [ffmpeg.4167] Process ffmpeg with PID 4167 stopped with returncode 255
2024-10-01 09:12:40.732 DEBUG (MainThread) [music_assistant.audio.media_stream] stream aborted (with code 255) for apple_music--4hmSTk5D://track/1690644641 - seconds streamed: 20.0
2024-10-01 09:12:40.732 VERBOSE (MainThread) [ffmpeg.4166] Process ffmpeg with PID 4166 stopped with returncode 255
2024-10-01 09:12:40.913 VERBOSE (MainThread) [music_assistant.streams] Got GET request to /single/media_player.main_stereo/c412e4f4e67342bba045bdbb05cd4277.mp3 from 192.168.2.31
headers: <CIMultiDictProxy('User-Agent': 'ESP32 HTTP Client/1.0', 'Host': '192.168.1.2:8097', 'Content-Length': '0')>

2024-10-01 09:12:40.961 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem Sutton - Oceans Apart (library://track/18398) to Main Stereo
2024-10-01 09:12:40.969 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4182
2024-10-01 09:12:40.969 VERBOSE (MainThread) [ffmpeg.4182] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -ac 2 -channel_layout stereo -ar 44100 -acodec pcm_f32le -f f32le -i - -af alimiter=limit=-2dB:level=false:asc=true,aresample=resampler=soxr:precision=30:osf=s16:dither_method=triangular_hp -f mp3 -b:a 320k -
2024-10-01 09:12:40.970 DEBUG (MainThread) [music_assistant.audio.media_stream] start media stream for: apple_music--4hmSTk5D://track/1651724072
2024-10-01 09:12:40.981 VERBOSE (MainThread) [ffmpeg] Process ffmpeg started with PID 4183
2024-10-01 09:12:40.982 VERBOSE (MainThread) [ffmpeg.4183] started with args: -hide_banner -loglevel info -nostats -ignore_unknown -protocol_whitelist file,hls,http,https,tcp,tls,crypto,pipe,data,fd,rtp,udp -decryption_key 14f8b4e0d9858bb9dd5b789a2fcad1ce -reconnect 1 -reconnect_streamed 1 -reconnect_on_network_error 1 -reconnect_on_http_error 5xx,4xx -i <URL> -af volume=-1.05dB -acodec pcm_f32le -f f32le -ar 44100 -ac 2 -
2024-10-01 09:12:42.037 VERBOSE (MainThread) [ffmpeg.4183] Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'https://aod-ssl.itunes.apple.com/itunes-assets/Music122/v4/ac/61/08/ac610843-17d0-b4dc-cdea-c3e15626464e/mzaf_A1651724072.rmhq.aac.wa.mp4':
2024-10-01 09:12:42.037 VERBOSE (MainThread) [ffmpeg.4183] Metadata:
2024-10-01 09:12:42.037 VERBOSE (MainThread) [ffmpeg.4183] major_brand     : mp42
2024-10-01 09:12:42.037 VERBOSE (MainThread) [ffmpeg.4183] minor_version   : 1
2024-10-01 09:12:42.038 VERBOSE (MainThread) [ffmpeg.4183] compatible_brands: mp41mp42isomhlsf
2024-10-01 09:12:42.038 VERBOSE (MainThread) [ffmpeg.4183] creation_time   : 2022-10-26T22:28:31.000000Z
2024-10-01 09:12:42.038 VERBOSE (MainThread) [ffmpeg.4183] Duration: 00:01:31.46, start: 0.000000, bitrate: 263 kb/s
2024-10-01 09:12:42.039 VERBOSE (MainThread) [ffmpeg.4183] Stream #0:0[0x1](und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 259 kb/s (default)
2024-10-01 09:12:42.040 DEBUG (MainThread) [ffmpeg.4183] Detected (input) content type: aac (aac)
2024-10-01 09:12:42.040 VERBOSE (MainThread) [ffmpeg.4183] Metadata:
2024-10-01 09:12:42.040 VERBOSE (MainThread) [ffmpeg.4183] creation_time   : 2022-10-26T22:28:31.000000Z
2024-10-01 09:12:42.040 VERBOSE (MainThread) [ffmpeg.4183] handler_name    : Core Media Audio
2024-10-01 09:12:42.041 VERBOSE (MainThread) [ffmpeg.4183] vendor_id       : [0][0][0][0]
2024-10-01 09:12:42.041 VERBOSE (MainThread) [ffmpeg.4183] Side data:
2024-10-01 09:12:42.042 VERBOSE (MainThread) [ffmpeg.4183] unknown side data type 24 (488 bytes)
2024-10-01 09:12:42.043 VERBOSE (MainThread) [ffmpeg.4183] Stream mapping:
2024-10-01 09:12:42.043 VERBOSE (MainThread) [ffmpeg.4183] Stream #0:0 -> #0:0 (aac (native) -> pcm_f32le (native))
2024-10-01 09:12:42.043 VERBOSE (MainThread) [ffmpeg.4183] Press [q] to stop, [?] for help
2024-10-01 09:12:42.051 VERBOSE (MainThread) [ffmpeg.4183] Output #0, f32le, to 'pipe:':
2024-10-01 09:12:42.051 VERBOSE (MainThread) [ffmpeg.4183] Metadata:
2024-10-01 09:12:42.051 VERBOSE (MainThread) [ffmpeg.4183] major_brand     : mp42
2024-10-01 09:12:42.052 VERBOSE (MainThread) [ffmpeg.4183] minor_version   : 1
2024-10-01 09:12:42.052 VERBOSE (MainThread) [ffmpeg.4183] compatible_brands: mp41mp42isomhlsf
2024-10-01 09:12:42.052 VERBOSE (MainThread) [ffmpeg.4183] encoder         : Lavf60.16.100
2024-10-01 09:12:42.052 VERBOSE (MainThread) [ffmpeg.4183] Stream #0:0(und): Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s (default)
2024-10-01 09:12:42.052 VERBOSE (MainThread) [ffmpeg.4183] Metadata:
2024-10-01 09:12:42.054 VERBOSE (MainThread) [ffmpeg.4183] creation_time   : 2022-10-26T22:28:31.000000Z
2024-10-01 09:12:42.055 VERBOSE (MainThread) [ffmpeg.4183] handler_name    : Core Media Audio
2024-10-01 09:12:42.055 VERBOSE (MainThread) [ffmpeg.4183] vendor_id       : [0][0][0][0]
2024-10-01 09:12:42.055 VERBOSE (MainThread) [ffmpeg.4183] encoder         : Lavc60.31.102 pcm_f32le
2024-10-01 09:12:42.055 VERBOSE (MainThread) [ffmpeg.4183] Side data:
2024-10-01 09:12:42.055 VERBOSE (MainThread) [ffmpeg.4183] unknown side data type 24 (488 bytes)
2024-10-01 09:12:42.116 VERBOSE (MainThread) [ffmpeg.4182] Input #0, f32le, from 'fd:':
2024-10-01 09:12:42.116 VERBOSE (MainThread) [ffmpeg.4182] Duration: N/A, bitrate: 2822 kb/s
2024-10-01 09:12:42.117 VERBOSE (MainThread) [ffmpeg.4182] Stream #0:0: Audio: pcm_f32le, 44100 Hz, stereo, flt, 2822 kb/s
2024-10-01 09:12:42.122 VERBOSE (MainThread) [ffmpeg.4182] Stream mapping:
2024-10-01 09:12:42.122 VERBOSE (MainThread) [ffmpeg.4182] Stream #0:0 -> #0:0 (pcm_f32le (native) -> mp3 (libmp3lame))
2024-10-01 09:12:42.157 VERBOSE (MainThread) [ffmpeg.4182] Output #0, mp3, to 'pipe:':
2024-10-01 09:12:42.157 VERBOSE (MainThread) [ffmpeg.4182] Metadata:
2024-10-01 09:12:42.158 VERBOSE (MainThread) [ffmpeg.4182] TSSE            : Lavf60.16.100
2024-10-01 09:12:42.158 VERBOSE (MainThread) [ffmpeg.4182] Stream #0:0: Audio: mp3, 44100 Hz, stereo, s16p, 320 kb/s
2024-10-01 09:12:42.158 VERBOSE (MainThread) [ffmpeg.4182] Metadata:
2024-10-01 09:12:42.158 VERBOSE (MainThread) [ffmpeg.4182] encoder         : Lavc60.31.102 libmp3lame
2024-10-01 09:12:42.245 VERBOSE (MainThread) [ffmpeg.4183] Process ffmpeg with PID 4183 stopped with returncode 255
2024-10-01 09:12:42.245 DEBUG (MainThread) [music_assistant.audio.media_stream] stream aborted (with code 255) for apple_music--4hmSTk5D://track/1651724072 - seconds streamed: 4.0
2024-10-01 09:12:42.254 VERBOSE (MainThread) [ffmpeg.4182] Process ffmpeg with PID 4182 stopped with returncode 255
```

</p>
</details>

**Comment by johnboiles** [comment]
The workaround that solved this for me was to disable `Queue Flow mode`. But that option was removed in ~2.2.3. The downside of course is a few seconds gap between songs, but that's definitely preferable to this issue.

Would y'all consider bringing back the option for disabling queue flow mode even for players that don't support enqueue next. At least until all the queue flow mode related issues (e.g. #2893, #2940) are worked out?

**Comment by GSVil** [comment]
I have this issue (and others) with my ESPHome Media Players, Songs skipping over and over. Also resume playing after broadcast not working, broadcasts repeating over and over. I usually have to restart Music Assistant Server, or reboot the esp32 to get stability back. If I look at the esp32 log it appears that Music Assistant keeps sending new streams every second. With the resume issue I get a 60sec timeout error in the music assistant logs, probably because the Media Player status never goes to Idle. If you tell me which logs to turn on I'll gladly do so. I don't get any of these issues when using Google Speakers, or my SqueezeLite ESP32's.

**Comment by michguy2000** [comment]
Hello.

The problem still persists, but it is sporadic now.  Twice last week when MA is executed via a script, it played only a few seconds of each song.  When I stopped MA and cleared the queue and then executed the playlist manually, it would work without issue.

 

But like I said the problem is not consistent now.  Yesterday and today, the MA executed via script and played without issue.

Regards

**Comment by OzGav** [comment]
@michguy2000 This thread is getting long with others chiming in so please confirm what player provider this is affecting. Have you tried other player providers to confirm this is isolated to just this one type?

**Comment by OzGav** [comment]
Please imstall the snapcast provider and try playing to that to see if the problem occurs there

**Comment by OzGav** [comment]
@michguy2000 firstly just to make this manageable when you respond via email can you you remove any quoted text in your reply to this so it remains readable. I have edited your responses above to remove all of that text.

ok back to it. Can I just confirm my understanding. You are using cast devices. You have added these cast devices directly to MA using the native cast provider where they were automatically discovered . You also have them in groups. You have grouped them in the Google Home App and this was also automatically discovered by MA. Then all music playback works fine from local provider and YTM except for playlists. These playlists contain with all local music tracks or all YTM tracks. The YTM playlist was created on YouTube. 

What I have written above is a standard setup. Please check it carefully and advise if it is all correct?

**Comment by michguy2000** [comment]
Can I just confirm my understanding. You are using cast devices.  - Yes. I only have Google/Nest speakers for casting.

You have added these cast devices directly to MA using the native cast provider where they were automatically discovered . You also have them in groups. You have grouped them in the Google Home App and this was also automatically discovered by MA.  – Yes. Correct.

Then all music playback works fine from local provider and YTM except for playlists. These playlists contain with all local music tracks or all YTM tracks. The YTM playlist was created on YouTube.

Yes and No.  Yes, the YTM playlist were created on YouTube Music.  YTM is not doing the skip track error.  It will playback the song entirely.  However I did find a problem that it appears that Google made some changes so none of my music on YouTube music that was uploaded to YTM will not cast anymore.  This has nothing to do with MA.  When YTM reaches a song in the playlist that I uploaded from my library, playback stops when casting, but will play if executed from my PC.  Just mentioning this in case there are others that have this issue. 

I just tested again with MA while composing this email and the skip problem did not occur.  Playlist from my local NAS and also from YTM (non-uploaded music) worked without issue.  When I first reported this, it was a problem I could always duplicate, but not now.  Not since last week. 

Regards

**Comment by OzGav** [comment]
If you can’t reproduce the problem anymore then is this fixed?

**Comment by michguy2000** [comment]
If you can’t reproduce the problem anymore then is this fixed? – 

It hasn’t happened again since the week of October 13th so yes, I guess we can close the issue.  Not sure what changed that corrected it, but thank you for your assistance.

Regards

**Comment by johnboiles** [comment]
Happened to me last week

**Comment by OzGav** [comment]
@johnboiles have you seen it on the latest version?

**Comment by michguy2000** [comment]
I tried installing Music Assistant Server 2.3.1 yesterday.   It didn’t work.  

Even after uninstalling MA and doing a fresh install using 2.3.1, MA would start but the web Ui screen would be blank with just a circle spinning.

I got frustrated and just restored Home Assistant using the previous night’s backup.

**Comment by OzGav** [comment]
@michguy2000 please open a new issue

**Comment by johnboiles** [comment]
I had the same thing on 2.3.1. It wouldn't load and I gave up. I suspect it was resource constrained while doing the initial syncing and so would never fully serve the front end

**Comment by OzGav** [comment]
Can you guys try 2.4.0b2

**Comment by michguy2000** [comment]
Hi.  I’m not running the Music Assistant Server (beta).  It’s the Music Assistant Server 2.2.9.

 

Can I just stop 2.2.9 and install the beta 2.4.0b2?

Would I have to set everything in MA up again?  Music providers, Player providers, Players?

Or will the 2.4.0b2 automatically use the current MA setup?

 

Thanks

**Comment by OzGav** [comment]
Yes you can stop the stable and run the beta in parallel.

Unfortunately, you do need to setup everything again as the containers are completely separate. It would be helpful for us for you to do so so we can tell if this problem is fixed for you. Others who reported on discord have found the issue resolved. You can go back to stable once the fix is pushed to there

**Comment by michguy2000** [comment]
Okay.  I installed the Beta version 2.4.0b2 and went through the entire MA config.  I ran a couple of tests and all the songs in several playlist worked without skipping.  I’m about to leave the house so when I come back, the music automation/scripts should start.  I’ll send a follow-up email with the results later.

 

Side note:  Other than Google/Nest speakers, what brands are you all using to cast music from MA?

Thanks

**Comment by OzGav** [comment]
2.3.2 has been released which fixes the resource hogging issue.

**Comment by OzGav** [comment]
@michguy2000 If you literally mean Cast I only use Google/Nest speakers. Otherwise I have some Airplay, some slimproto, and Snapcast.

**Comment by michguy2000** [comment]
Thank you for your assistance.  I've been stable on the BETA release now for a week.  You can close this issue.
Regards
